### ISSUE TITLE
P3: Document that LFM2 expert bias changes REAP metric semantics vs the original paper

### ISSUE THREAD

**Issue by egesabanci** [egesabanci/reap-mlx#46]
## Summary

When `use_expert_bias=True` (LFM2 models), the router adds expert bias to softmax outputs before top-k selection. These biased scores flow into `PruningState.accumulate()`, meaning `weighted_ean_sum` and the REAP metric incorporate the bias.

This is correct behavior for matching actual model output (the MoE computation uses the same biased scores), but it diverges from the original REAP paper formulation which assumes pure softmax probabilities as weights.

## Location

`src/reap/router.py`, lines ~211-213:
```python
gates = mx.softmax(logits, axis=-1)
if self.use_expert_bias:
    gates = gates + self.expert_bias
```

These biased gates become `routing.scores` in `RouterResult`, which then flow into:
```python
state.accumulate(
    indices=routing.indices,
    scores=routing.scores.astype(mx.float32),   # <-- biased scores
    selected_outputs=selected_outputs.astype(mx.float32),
)
```

The REAP metric (`weighted_ean_sum / count_denominator`) then reflects bias-weighted activations, not pure softmax-weighted activations.

## Proposed fix

Add a note in the README Pruning Methods section and/or in `Lfm2MoeRouter` docstring:

> When `use_expert_bias=True` (LFM2 models), the REAP saliency score incorporates
> the expert bias in the activation weighting. This matches the actual model output
> computation but may produce different rankings than the original paper formulation
> which assumes pure softmax probability weights.

**Effort**: 3-5 lines of documentation.
